This patch works around an issue with vim dumping core with SIGSEGV when
handling a more benign signal (typically SIGHUP or SIGTERM).  Vim tries to
run all the exit-vim autocommands, which end up busting the alternate stack
at its default size.  Doubling the default seems to do the trick, but it
continues to be just a guess.

This fix was suggested by the maintainer, but the patch has not yet been
submitted upstream.

--- a/src/os_unix.c	2018-07-19 03:55:59.882437014 +0000
+++ b/src/os_unix.c	2018-07-19 03:50:04.638443666 +0000
@@ -912,9 +912,10 @@ mch_stackcheck(char *p)
  * completely full.
  */
 
-#ifndef SIGSTKSZ
-# define SIGSTKSZ 8000    /* just a guess of how much stack is needed... */
+#ifdef SIGSTKSZ
+# undef SIGSTKSZ
 #endif
+#define SIGSTKSZ 16384    /* just a guess of how much stack is needed... */
 
 # ifdef HAVE_SIGALTSTACK
 static stack_t sigstk;			/* for sigaltstack() */
